plugins/module_utils/network/sonic/config/bgp_neighbors_af/bgp_neighbors_af.py

```diff
@@ -274,11 +355,78 @@ def _state_deleted(self, want, have, diff):
 
         return commands, requests
 
+    def remove_empty_neighbors(self, have):
+        new_conf = []
+        for conf in have:
+            temp_conf, temp_nbr = {}, []
+            for nbr in conf.get('neighbors', []):
+                if 'address_family' in nbr:
+                    temp_nbr.append(nbr)
+            if temp_nbr:
+                temp_conf['bgp_as'] = conf['bgp_as']
+                temp_conf['vrf_name'] = conf['vrf_name']
+                temp_conf['neighbors'] = temp_nbr
+                new_conf.append(temp_conf)
+        return new_conf
+
     def set_val(self, cfg, var, src_key, des_key):
         value = var.get(src_key, None)
         if value is not None:
             cfg[des_key] = value
 
+    def _get_replaced_config(self, want, have):
+        add_config, del_config = [], []
+
+        diff1 = get_diff(want, have, TEST_KEYS)
+        diff2 = get_diff(have, want, TEST_KEYS)
+        for cmd in diff1:
+            del_cfg, add_cfg = {}, {}
+            vrf_name = cmd.get('vrf_name')
+            bgp_as = cmd.get('bgp_as')
+            match = next((cfg for cfg in diff2 if (cfg['vrf_name'] == vrf_name and (cfg['bgp_as'] == bgp_as))), None)
+
+            if match:
+                neighbors = cmd.get('neighbors', [])
```

Comment:
If not match, "cmd" will not be added to "add_config"? So that it will not be configured to remote system?
Same is for the following "neighbor" and "address_family".

Please check and verify!

Please add regression test cases to test those cases.

Reply:
Created a new test_case_replace_02 to check this.

Handled both "not match" and "not match_nbr".